ethcore/sync/src/chain/supplier.rs

```diff
@@ -135,7 +136,7 @@ impl SyncSupplier {
 		let inc = (skip + 1) as BlockNumber;
 		let overlay = io.chain_overlay().read();
 
-		while number <= last && count < max_count {
+		while (number <= last || overlay.contains_key(&number)) && count < max_count {
```

Comment:
Is this change unrelated to the PR? Maybe a comment would help.

Reply:
Yep kind of unrelated : it actually fixes a bug where you wouldn't send the _ForkBlock_ header before being synced even if we cached it for this exact reason : we want to share it with other node ASAP so that they can confirm we are on the right fork.